java-checks/src/test/java/org/sonar/java/checks/PackageInfoCheckTest.java

```diff
@@ -19,32 +19,57 @@
  */
 package org.sonar.java.checks;
 
+import com.sonar.sslr.api.RecognitionException;
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Set;
 import org.junit.Test;
+import org.sonar.api.batch.fs.internal.TestInputFileBuilder;
+import org.sonar.api.batch.sensor.internal.SensorContextTester;
+import org.sonar.api.config.internal.MapSettings;
+import org.sonar.api.internal.SonarRuntimeImpl;
+import org.sonar.api.utils.Version;
+import org.sonar.java.SonarComponents;
 import org.sonar.java.ast.JavaAstScanner;
 import org.sonar.java.model.VisitorsBridge;
 
-import java.io.File;
-import java.util.Set;
-
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class PackageInfoCheckTest {
 
   @Test
   public void test() throws Exception {
     PackageInfoCheck check = new PackageInfoCheck();
-    JavaAstScanner.scanSingleFileForTests(new File("src/test/files/checks/packageInfo/HelloWorld.java"), new VisitorsBridge(check));
+    File file = new File("src/test/files/checks/packageInfo/HelloWorld.java");
+    JavaAstScanner.scanSingleFileForTests(file, new VisitorsBridge(Collections.singletonList(check), Collections.emptyList(), sonarComponents(file)));
     assertThat(check.directoriesWithoutPackageFile).isEmpty();
   }
 
   @Test
   public void testNoPackageInfo() throws Exception {
     PackageInfoCheck check = new PackageInfoCheck();
-    JavaAstScanner.scanSingleFileForTests(new File("src/test/files/checks/packageInfo/nopackageinfo/nopackageinfo.java"), new VisitorsBridge(check));
-    JavaAstScanner.scanSingleFileForTests(new File("src/test/files/checks/packageInfo/nopackageinfo/HelloWorld.java"), new VisitorsBridge(check));
+    File file = new File("src/test/files/checks/packageInfo/nopackageinfo/nopackageinfo.java");
+    JavaAstScanner.scanSingleFileForTests(file, new VisitorsBridge(Collections.singletonList(check), Collections.emptyList(), sonarComponents(file)));
+    file = new File("src/test/files/checks/packageInfo/nopackageinfo/HelloWorld.java");
+    JavaAstScanner.scanSingleFileForTests(file, new VisitorsBridge(Collections.singletonList(check), Collections.emptyList(), sonarComponents(file)));
     Set<File> set = check.directoriesWithoutPackageFile;
     assertThat(set).hasSize(1);
     assertThat(set.iterator().next().getName()).isEqualTo("nopackageinfo");
   }
 
+  static SonarComponents sonarComponents(File file) {
+    SensorContextTester context = SensorContextTester.create(new File("")).setRuntime(SonarRuntimeImpl.forSonarLint(Version.create(6, 7)));
+    context.setSettings(new MapSettings().setProperty("sonar.java.failOnException", true));
+    SonarComponents sonarComponents = new SonarComponents(null, context.fileSystem(), null, null, null) {
+      @Override
+      public boolean reportAnalysisError(RecognitionException re, File file) {
+        return false;
```

Comment:
This feels like a workaround to avoid having to properly define the `filesystem` and `context` objects for the tests.
I thing it would be much clearer to understand something like:
```
    File moduleBaseDir = new File("");
    SensorContextTester context = SensorContextTester.create(moduleBaseDir);
    context.fileSystem().setWorkDir(moduleBaseDir.toPath());
    SonarComponents sonarComponents = new SonarComponents(null, context.fileSystem(), null, null, null);
    sonarComponents.setSensorContext(context);

    DefaultInputFile inputFile = new TestInputFileBuilder("", moduleBaseDir, file).setCharset(StandardCharsets.UTF_8)
      .setContents(new String(Files.readAllBytes(file.toPath()))).build();
    context.fileSystem().add(inputFile);
```